pkg/sysregistriesv2/system_registries_v2_test.go

```diff
@@ -671,6 +671,122 @@ func TestPullSourcesFromReference(t *testing.T) {
 	assert.Equal(t, 1, len(pullSources))
 }
 
+func TestPullSourcesMirrorFromReference(t *testing.T) {
+	sys := &types.SystemContext{
+		SystemRegistriesConfPath:    "testdata/pull-sources-mirror-reference.conf",
+		SystemRegistriesConfDirPath: "testdata/this-does-not-exist",
+	}
+	registries, err := GetRegistries(sys)
+	require.NoError(t, err)
+	assert.Equal(t, 7, len(registries))
+
+	digest := "@sha256:aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+	tag := ":aaa"
+	for _, tc := range []struct {
+		registry      string
+		digestSources []string
+		tagSources    []string
+	}{
+		// Registry A has mirrors allow any kind of pull
+		{
+			"registry-a.com/foo/image",
+			[]string{"mirror-1.registry-a.com", "mirror-2.registry-a.com", "registry-a.com/bar"},
+			[]string{"mirror-1.registry-a.com", "mirror-2.registry-a.com", "registry-a.com/bar"},
+		},
+		// Registry B has mirrors allow digests pull only
+		{
+			"registry-b.com/foo/image",
+			[]string{"mirror-1.registry-b.com", "mirror-2.registry-b.com", "registry-b.com/bar"},
+			[]string{"registry-b.com/bar"},
+		},
+		// Registry C has a mirror allows digest pull only and a mirror allows any kind of pull
+		{
+			"registry-c.com/foo/image",
+			[]string{"mirror-1.registry-c.com", "mirror-2.registry-c.com", "registry-c.com/bar"},
+			[]string{"mirror-1.registry-c.com", "registry-c.com/bar"},
+		},
+		// Registry D set digest-only for registry level, allows only digest pulls
+		// Registry D has no digest-only set for mirrors table
+		{
+			"registry-d.com/foo/image",
+			[]string{"mirror-1.registry-d.com", "mirror-2.registry-d.com", "registry-d.com/bar"},
+			[]string{"registry-d.com/bar"},
+		},
+		// Registry E has mirrors only allows tag pull
+		{
+			"registry-e.com/foo/image",
+			[]string{"registry-e.com/bar"},
+			[]string{"mirror-1.registry-e.com", "mirror-2.registry-e.com", "registry-e.com/bar"},
+		},
+		// Registry F has one tag only mirror does not allow digest pull
+		{
+			"registry-f.com/foo/image",
+			[]string{"mirror-1.registry-f.com", "registry-f.com/bar"},
+			[]string{"mirror-1.registry-f.com", "mirror-2.registry-f.com", "registry-f.com/bar"},
+		},
+		// Registry G has one digest-only pull and one tag only pull
+		{
+			"registry-g.com/foo/image",
+			[]string{"mirror-1.registry-g.com", "mirror-3.registry-g.com", "mirror-4.registry-g.com", "registry-g.com/bar"},
+			[]string{"mirror-2.registry-g.com", "mirror-3.registry-g.com", "mirror-4.registry-g.com", "registry-g.com/bar"},
+		},
+	} {
+		// Digest
+		digestedRef := toNamedRef(t, tc.registry+digest)
+		registry, err := FindRegistry(sys, digestedRef.Name())
+		require.NoError(t, err)
+		assert.NotNil(t, registry)
```

Comment:
Non-blocking: `require.NotNil`, otherwise the next line would crash.

The test fails either way, so it can be fixed up if it ever happens.

Same in the `taggedRef` case below.